### DISCUSSION TITLE
Query Assistance

### DISCUSSION THREAD

**Discussion by stsree** [Gremlinq/ExRam.Gremlinq#367] [Q&A]
Hi,
I am trying to convert the below query to Gremlinq. Could anyone help convert?

```
g.V('efae9a6d-1228-486b-bb8d-afbc22fd5f9a')
.outE('With').inV().hasLabel('Area').as('t')
.outE('Covers').inV().hasLabel('State').as('s')
.inE('LivesIn').outV().hasLabel('Customer').as('p')
.choose(inE('Attends'), inE('Attends').outV(), __.project()).as('event')
.choose(select('p').inE('Issues'), inE('Issues').outV(), __.project()).as('vendor')
.select('p','s','vendor','event')
```

Thank you!

**Comment by danielcweber** [comment]
What do you already have? Where do you get stuck?

**Reply by stsree** [reply]
Thanks Dan. I tried below.
```
var users = await g.V<Rep>(repDetail.Id)
                                  .OutE<AssociatedWith>().InV<Area>().As((_, area) =>
                                 _.OutE<Covers>().InV<State>().As((_, state) =>
                                 _.InE<LivesIn>().OutV<Customer>().As((_, customer) => 
                                 _.Choose(c => _.InE<Attends>(), t => t.InE<Attends>().OutV(), f => f.InE<Attends>().OutV()).As((_, event) => 
                                 _.Select(customer, state, event)))));
```

As you can see, I haven't even added the last Choose query in this yet. For the above code, I am getting this exception.

`{"A query continuation must originate from the query that was passed to the continuation function. Did you accidentally use 'g' in the continuation?"}`

Any guidance/example on Choose and Project is much appreciated.

Thanks,
Sree

**Reply by stsree** [reply]
The purpose of the query is to get results some what similar to (ignore the condition details pls)

select customer 
          from area inner join state
          inner join customer
          left join(with customer) event
          left join(with customer) vendor

**Reply by danielcweber** [reply]
In Choose:;
`c => _...` when you meant `c => c ...`

**Reply by stsree** [reply]
If I try, 
`_.Choose(c => c.InE<Attends>(), t => t.InE<Attends>().OutV(), f => f.InE<Attends>().OutV()).As((_, event) => `
code is not compiling. I get the message as "Customer does not contain a definition for InE accepting a first argument of type Customer could be found".

**Reply by stsree** [reply]
Is there an option to execute the groovy query and bind the values back to object as well?

**Reply by danielcweber** [reply]
>If I try,
_.Choose(c => c.InE<Attends>(), t => t.InE<Attends>().OutV(), f => f.InE<Attends>().OutV()).As((_, event) => code is not compiling. 

Which Choose-method-signature are you aiming for? And what's the actual types of the parameters you're passing in?

>Is there an option to execute the groovy query and bind the values back to object as well?

Nope, unfortunately outside the scope of Gremlinq.
